Failed: opcap-weekly-audit-4.9-cm2ws Aug 12, 2022, 6:20 PM

I0812 23:20:17.271158       7 request.go:601] Waited for 1.042655851s due to client-side throttling, not priority and fairness, request: GET:https://api.audit-4-9.opdev.io:6443/apis/node.k8s.io/v1?timeout=32s
check called

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:21:15.886913123 +0000 UTC m=+64.716236776
OpenShift Version: 4.9.46
Package Name: redhat-marketplace-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: marketplaceconfigs package: redhat-marketplace-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:21:58.505211922 +0000 UTC m=+107.334535576
OpenShift Version: 4.9.46
Package Name: headspinunifiedcontroller
Channel: alpha
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: headspinunifiedcontrollers package: headspinunifiedcontroller

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:22:46.077989081 +0000 UTC m=+154.907312734
OpenShift Version: 4.9.46
Package Name: app-director-operator
Channel: 0.7.4
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand failed to create: clusterwideguvnors package: app-director-operator error: the server could not find the requested resource

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:23:07.856046651 +0000 UTC m=+176.685370306
OpenShift Version: 4.9.46
Package Name: hpe-csi-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: hpecsidrivers package: hpe-csi-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:23:56.549856507 +0000 UTC m=+225.379180161
OpenShift Version: 4.9.46
Package Name: dynatrace-operator
Channel: alpha
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand failed to create: dynakubes package: dynatrace-operator error: admission webhook "webhook.dynatrace.com" denied the request: 
1 error(s) found in the Dynakube
 1. The DynaKube's specification is missing the API URL or still has the example value set.
	Make sure you correctly specify the URL in your custom resource.
	

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:24:44.539131262 +0000 UTC m=+273.368454922
OpenShift Version: 4.9.46
Package Name: cloudbees-ci
Channel: alpha
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: cloudbeescis package: cloudbees-ci

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:25:20.18173656 +0000 UTC m=+309.011060218
OpenShift Version: 4.9.46
Package Name: rookout-k8s-operator
Channel: alpha
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand failed to create: rookouts package: rookout-k8s-operator error: Rookout.rookout.rookout.com "rookout-operator-configuration" is invalid: spec.matchers.labels: Invalid value: "string": spec.matchers.labels in body must be of type object: "string"

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:26:16.802249166 +0000 UTC m=+365.631572820
OpenShift Version: 4.9.46
Package Name: citrix-cpx-with-ingress-controller-operator
Channel: alpha
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: citrixcpxwithingresscontrollers package: citrix-cpx-with-ingress-controller-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:26:40.476169847 +0000 UTC m=+389.305493506
OpenShift Version: 4.9.46
Package Name: xcrypt-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: xcrypts package: xcrypt-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:27:17.103326092 +0000 UTC m=+425.932649750
OpenShift Version: 4.9.46
Package Name: ovms-operator
Channel: alpha
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: modelservers package: ovms-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:28:02.96136366 +0000 UTC m=+471.790687318
OpenShift Version: 4.9.46
Package Name: f5-bigip-ctlr-operator
Channel: beta
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: f5bigipctlrs package: f5-bigip-ctlr-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:28:54.630852925 +0000 UTC m=+523.460176578
OpenShift Version: 4.9.46
Package Name: tenant-operator
Channel: alpha
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand failed to create: tenants package: tenant-operator error: the server could not find the requested resource

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:29:18.591240753 +0000 UTC m=+547.420564407
OpenShift Version: 4.9.46
Package Name: zookeeper-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: zookeeperclusters package: zookeeper-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:29:42.220273994 +0000 UTC m=+571.049597651
OpenShift Version: 4.9.46
Package Name: cilium
Channel: 1.12
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: ciliumconfigs package: cilium

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:30:18.895011592 +0000 UTC m=+607.724335245
OpenShift Version: 4.9.46
Package Name: kubeturbo-certified
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: kubeturbos package: kubeturbo-certified

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:30:50.584300639 +0000 UTC m=+639.413624293
OpenShift Version: 4.9.46
Package Name: zabbix-operator-certified
Channel: lts
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: zabbixagents package: zabbix-operator-certified

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:31:52.455268996 +0000 UTC m=+701.284592650
OpenShift Version: 4.9.46
Package Name: installkit-customer
Channel: stable
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: timeout
Message: installing: waiting for deployment runai-operator to become ready: deployment "runai-operator" not available: Deployment does not have minimum availability.
Reason: InstallWaiting
-----------------------------------------
exiting OperandInstall since no ALM_Examples found in CSV

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:32:13.512302521 +0000 UTC m=+722.341626176
OpenShift Version: 4.9.46
Package Name: runtime-component-operator-certified
Channel: beta2
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: runtimecomponents package: runtime-component-operator-certified

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:32:36.34524332 +0000 UTC m=+745.174566974
OpenShift Version: 4.9.46
Package Name: open-liberty-certified
Channel: beta2
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: openlibertyapplications package: open-liberty-certified

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:33:18.088937974 +0000 UTC m=+786.918261628
OpenShift Version: 4.9.46
Package Name: instana-agent-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: issuers package: instana-agent-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:33:44.586932848 +0000 UTC m=+813.416256502
OpenShift Version: 4.9.46
Package Name: ibm-application-gateway-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: ibmapplicationgateways package: ibm-application-gateway-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:34:10.455256871 +0000 UTC m=+839.284580523
OpenShift Version: 4.9.46
Package Name: iomesh-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
exiting OperandInstall since no ALM_Examples found in CSV

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:34:39.519249239 +0000 UTC m=+868.348572893
OpenShift Version: 4.9.46
Package Name: couchbase-enterprise-certified
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: couchbaseclusters package: couchbase-enterprise-certified

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:35:04.802275122 +0000 UTC m=+893.631598775
OpenShift Version: 4.9.46
Package Name: nxiq-operator-certified
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: nexusiqs package: nxiq-operator-certified

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:35:28.647407415 +0000 UTC m=+917.476731078
OpenShift Version: 4.9.46
Package Name: antrea-operator-for-kubernetes
Channel: alpha
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand failed to create: antreainstalls package: antrea-operator-for-kubernetes error: AntreaInstall.operator.antrea.vmware.com "antrea-install" is invalid: spec.antreaPlatform: Required value

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:35:53.687189896 +0000 UTC m=+942.516513549
OpenShift Version: 4.9.46
Package Name: percona-xtradb-cluster-operator-certified
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: perconaxtradbclusters package: percona-xtradb-cluster-operator-certified

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:36:37.891253862 +0000 UTC m=+986.720577514
OpenShift Version: 4.9.46
Package Name: acc-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: astracontrolcenters package: acc-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:37:23.338919788 +0000 UTC m=+1032.168243441
OpenShift Version: 4.9.46
Package Name: nutanixcsioperator
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: nutanixcsistorages package: nutanixcsioperator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:38:00.406157706 +0000 UTC m=+1069.235481360
OpenShift Version: 4.9.46
Package Name: entando-k8s-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: entandodebundles package: entando-k8s-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:38:23.858390451 +0000 UTC m=+1092.687714109
OpenShift Version: 4.9.46
Package Name: xspc-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: xspcs package: xspc-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:39:00.089217655 +0000 UTC m=+1128.918541308
OpenShift Version: 4.9.46
Package Name: ibm-spectrum-symphony-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: OwnNamespace
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: symphonyclusters package: ibm-spectrum-symphony-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:40:00.341262792 +0000 UTC m=+1189.170586450
OpenShift Version: 4.9.46
Package Name: pachyderm-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: pachyderms package: pachyderm-operator

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:41:03.003259 +0000 UTC m=+1251.832582656
OpenShift Version: 4.9.46
Package Name: infoscale-licensing-operator
Channel: stable
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: timeout
Message: installing: waiting for deployment infoscale-licensing-operator to become ready: deployment "infoscale-licensing-operator" not available: Deployment does not have minimum availability.
Reason: InstallWaiting
-----------------------------------------
exiting OperandInstall since CSV has failed

opcap report:
-----------------------------------------
Report Date: 2022-08-12 23:42:41.091333013 +0000 UTC m=+1349.920656668
OpenShift Version: 4.9.46
Package Name: infinibox-operator-certified
Channel: stable
Catalog Source: certified-operators
Install Mode: AllNamespaces
Result: Succeeded
Message: install strategy completed with no errors
Reason: InstallSucceeded
-----------------------------------------
operand succeeded: infiniboxcsidrivers package: infinibox-operator-certified
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x15a971f]

goroutine 1 [running]:
github.com/opdev/opcap/internal/operator.operatorClient.GetCompletedCsvWithTimeout({{_, _}}, {_, _}, _)
	/home/mrhillsman/Development/goland/opcap/internal/operator/csv.go:60 +0x3df
github.com/opdev/opcap/internal/capability.(*CapAudit).OperatorInstall(0xc000180b00)
	/home/mrhillsman/Development/goland/opcap/internal/capability/operator_install.go:36 +0x3be
reflect.Value.call({0x1804be0, 0xc000180b00, 0x3}, {0x197fbb0, 0x4}, {0x0, 0x0, 0x1c57ef0})
	/usr/local/go/src/reflect/value.go:556 +0x845
reflect.Value.Call({0x1804be0, 0xc000180b00, 0x2e}, {0x0, 0x0, 0x0})
	/usr/local/go/src/reflect/value.go:339 +0xc5
github.com/opdev/opcap/internal/capability.(*capAuditor).RunAudits(0xc000771d68)
	/home/mrhillsman/Development/goland/opcap/internal/capability/auditor.go:84 +0x1bd
github.com/opdev/opcap/cmd.glob..func2(0x2997f80, {0x19800b8, 0x2, 0x2})
	/home/mrhillsman/Development/goland/opcap/cmd/check.go:79 +0x112
github.com/spf13/cobra.(*Command).execute(0x2997f80, {0xc0004866c0, 0x2, 0x2})
	/home/mrhillsman/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0x2998200)
	/home/mrhillsman/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/mrhillsman/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/opdev/opcap/cmd.Execute()
	/home/mrhillsman/Development/goland/opcap/cmd/root.go:32 +0x25
main.main()
	/home/mrhillsman/Development/goland/opcap/main.go:10 +0x17

--- a/cmd/check.go
+++ b/cmd/check.go
@@ -25,7 +25,7 @@ var checkCmd = &cobra.Command{
 	// TODO: provide Short description for check command
 	Short: "Checks if operator meets minimum capability requirement.",
 	// TODO: provide Long description for check command
-	Long:  `The 'check' command checks if OpenShift operators meet minimum
+	Long: `The 'check' command checks if OpenShift operators meet minimum
 requirements for Operator Capabilities Level to attest operator
 advanced features by running custom resources provided by CSVs
 and/or users.

--- a/internal/operator/csv.go
+++ b/internal/operator/csv.go
@@ -15,13 +15,13 @@ import (
 func (c operatorClient) GetCompletedCsvWithTimeout(namespace string, delay time.Duration) (operatorv1alpha1.ClusterServiceVersion, error) {
 
 	// csv will catch CSVs from watch events
-	var csv *operatorv1alpha1.ClusterServiceVersion
+	csv := &operatorv1alpha1.ClusterServiceVersion{}
 	var ok bool
 
 	// get watcher for csv
 	watcher, err := c.csvWatcher(namespace)
 	if err != nil {
-		return operatorv1alpha1.ClusterServiceVersion{}, err
+		return *csv, err
 	}
 
 	// eventChan receives all events from CSVs on the selected namespace

